README.md

```diff
@@ -1,7 +1,334 @@
-# S3-herald
+<p align="center">
+    <img src="https://raw.githubusercontent.com/PKief/vscode-material-icon-theme/ec559a9f6bfd399b82bb44393651661b08aaf7ba/icons/folder-markdown-open.svg" align="center" width="30%">
+</p>
+<p align="center"><h1 align="center">herald</h1></p>
+<p align="center">
+	<em>herald: Orchestrating object storage services</em>
+</p>
+<p align="center">
+	<img src="https://img.shields.io/github/license/expnt/herald?style=default&logo=opensourceinitiative&logoColor=white&color=0080ff" alt="license">
+	<img src="https://img.shields.io/github/last-commit/expnt/herald?style=default&logo=git&logoColor=white&color=0080ff" alt="last-commit">
+	<img src="https://img.shields.io/github/languages/top/expnt/herald?style=default&color=0080ff" alt="repo-top-language">
+	<img src="https://img.shields.io/github/languages/count/expnt/herald?style=default&color=0080ff" alt="repo-language-count">
+</p>
+<p align="center"><!-- default option, no dependency badges. -->
+</p>
+<p align="center">
+	<!-- default option, no dependency badges. -->
+</p>
+<br>
 
-An Simple and Light S3 proxy to mirrors that encompasses the following solutions.
+## Table of Contents
 
-- ...
+- [ Overview](#Overview)
+- [ Features](#Features)
+- [ Project Structure](#Project-Structure)
+- [ Getting Started](#Getting-Started)
+  - [ Prerequisites](#Prerequisites)
+  - [ Development](#Development)
+  - [ herald.yaml config file](#herald.yaml-config-file)
+  - [ Environment Variables](#Environment-Variables)
+  - [ Usage](#Usage)
+  - [ Testing](#Testing)
+- [ Project Roadmap](#Project-Roadmap)
+- [ Contributing](#Contributing)
+- [ Acknowledgments](#Acknowledgments)
 
-The proxy tries to cover the various [endpoints](https://docs.aws.amazon.com/AmazonS3/latest/API/API_Operations_Amazon_Simple_Storage_Service.html) that are available in an S3 Storage. The endpoints will be a blueprint during the proxy design.
+---
+
+## Overview
+
+The herald project is a proxy which sits on top of storage services that addresses the challenge of managing scalable cloud storage solutions by allowing you to connect with different types of object storages (S3, Swift) using the S3 protocol. The development was inspired due to some bugs in the OpenStack S3 proxy for the swift object storage service. The proxy includes other essential features besides fixing some of the bugs. It supports mirroring across different cloud storage providers, kubernetes friendly authentication scheme and others.
```

Comment:
still not convinced of the content, let's do it the human way? it does not have to be long, but rather clear, simple, straight to the goal, and include an example